File: solutions/110.balanced-binary-tree/balanced-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public boolean res = true;
    public boolean isBalanced(TreeNode root) {
        deep(root, 0);
        return res;
    }
    public int deep(TreeNode root, int n){
        if(root == null) return n;
        int left = deep(root.left, n + 1);
        int right = deep(root.right, n + 1);
        if(left -right > 1 || left -right < -1){
            res = false;
        }
        return Math.max(left, right);
    }
}

File: solutions/110.balanced-binary-tree/balanced-binary-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {boolean}
 */
var isBalanced = function(root) {
    let res = true;
    getHeight(root);
    return res;
    
    function getHeight(node){
        if(node === null) return 0;
        const left = getHeight(node.left);
        const right = getHeight(node.right);
        if(Math.abs(right - left) > 1){
            res = false;
        }
        return Math.max(left, right) + 1;
    }
};